fix(treesitter): keep tree view open when source window is still open

hyyker opened the pull request:
Problem: When there is a tree view opened by ```:InspectTree``` and the source buffer is open in multiple windows, closing one of the source windows will lead to the tree view being closed as well. This bug was introduced by #31181. Related issue: #31196

Solution: Check how many source windows are open when trying to quit one. If there are more than one, keep the tree view(s) open. If the only source window is closed, also close the tree view(s).

This should close #31196.

Commit:
fix(treesitter): keep tree view open when source window is still open

Problem: When there is a tree view opened by ```:InspectTree``` and
the source buffer is open in multiple windows, closing one of the
source windows will lead to the tree view being closed as well.
Related issue: #31196

Solution: Check how many source windows are open when trying to quit
one. If there are more than one, keep the tree view(s) open. If the
only source window is closed, also close the tree view(s).

lewis6991 commented on runtime/lua/vim/treesitter/dev.lua:
```diff
@@ -530,8 +530,13 @@ function M.inspect_tree(opts)
+
       -- close all tree windows
       for _, window in pairs(vim.fn.win_findbuf(b)) do
         close_win(window)
```
I think this now needs to `return true` to delete the autocmd.

hykerr replied:
I think I see what you mean. Changing the return to true will make it throw ```E855: Autocommands caused command to abort``` on the last window again which I was trying to fix. Do you think not returning true will have side effects?

zeertzjq replied:
You need to `return true` after all tree windows have been closed, not when the tree windows should be kept open.

hykerr replied:
Oh nvm you meant a a return at the end right? That works

Commit:
fix(treesitter): keep tree view open when source window is still open

Problem: When there is a tree view opened by :InspectTree and the
source buffer is open in multiple windows, closing one of the source
windows will lead to the tree view being closed as well. This bug
was introduced by #31181. Related issue: #31196

Solution: Check how many source windows are open when trying to
quit one. If there are more than one, keep the tree view(s) open.
If the only source window is closed, also close the tree view(s).

This should close #31196.
